Sources/XMLCoder/Box/Box.swift

```diff
@@ -0,0 +1,238 @@
+//
+//  Box.swift
+//  XMLCoder
+//
+//  Created by Vincent Esche on 12/17/18.
+//
+
+import Foundation
+
+internal enum Box {
+    case null(NullBox)
+    case bool(BoolBox)
+    case decimal(DecimalBox)
+    case signedInteger(SignedIntegerBox)
+    case unsignedInteger(UnsignedIntegerBox)
+    case floatingPoint(FloatingPointBox)
+    case string(StringBox)
+    case array(ArrayBox)
+    case dictionary(DictionaryBox)
+    
+    var isNull: Bool {
+        guard case .null(_) = self else {
+            return false
+        }
+        return true
+    }
+    
+    var isFragment: Bool {
+        switch self {
+        case .null(_): return true
+        case .bool(_): return true
+        case .decimal(_): return true
+        case .signedInteger(_): return true
+        case .unsignedInteger(_): return true
+        case .floatingPoint(_): return true
+        case .string(_): return true
+        case .array(_): return false
+        case .dictionary(_): return false
+        }
+    }
+    
+    var bool: BoolBox? {
+        guard case .bool(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var decimal: DecimalBox? {
+        guard case .decimal(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var signedInteger: SignedIntegerBox? {
+        guard case .signedInteger(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var unsignedInteger: UnsignedIntegerBox? {
+        guard case .unsignedInteger(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var floatingPoint: FloatingPointBox? {
+        guard case .floatingPoint(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var string: StringBox? {
+        guard case .string(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var array: ArrayBox? {
+        guard case .array(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var dictionary: DictionaryBox? {
+        guard case .dictionary(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var xmlString: String {
+        switch self {
```

Comment:
I think this `switch` could be avoided with a `Box` protocol as described in comments above.